### PR TITLE
Wait flag endpoint update

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will fix some issues with `wait_for_current_time_flag` and its use with endpoints.  

### Proposed changes

-  add some tests
- add error checks for multiple federates using the flag
- change the timing so wait_for_current_time works on time 0 as well
- make it so wait_for_current_time works with endpoints

### CODE REVIEW

**Review by nightlark — APPROVED** [review]

**Comment by nightlark** [comment]
It looks like there is an issue with `ftc_tests.timing_test1` consistently [failing](https://dev.azure.com/HELICS-test/HELICS/_build/results?buildId=9371&view=logs&j=93c97b79-db44-5fec-340e-53842163cada&t=70e148cc-01f4-56bc-cde3-8f079ab6897c&l=138) that should be resolved before merging.

**Comment by phlptp** [comment]
I fixed that test,  there are a few other tests I am trying to resolve as well before I would merge this, as well as the clang-tidy issues.

File: tests/helics/apps/TracerTests.cpp [hunk 1]
```diff
@@ -644,19 +644,20 @@ INSTANTIATE_TEST_SUITE_P(tracer_tests,
                          tracer_clone_file_tests,
                          ::testing::ValuesIn(simple_clone_test_files));
 
+/*
 #ifdef ENABLE_ZMQ_CORE
```
Comment (nightlark) [inline tests/helics/apps/TracerTests.cpp:648]: The #ifdef looks like it should be updated if the test is no longer using ZMQ.

Comment (phlptp) [reply tests/helics/apps/TracerTests.cpp:648]: I am playing with this test yet to see if it passes.  I will clean up once I figure what I want to do with it

**Pull request merged**